### DISCUSSION TITLE
Web UI for go-ios and maybe QVH

### DISCUSSION THREAD

**Discussion by danielpaulus** [danielpaulus/go-ios#158] [Ideas]
Continuing the discussion started in this issue:
https://github.com/danielpaulus/go-ios/issues/154

It was suggested to add a user interface to go-ios. I am open to the idea. 
If you are interested in describing your use case for such a UI please do that here. 
At the moment it seems like the most interest is in getting one of these "live testing"-remote control UIs going, that allows you to remotely interact with a device and see the screen.

**Comment by danielpaulus** [comment]
copy pasting what @shamanec  said:
Can't say anything about streaming, I spent some time trying to create an mjpeg stream using the screenshot functionality but I couldn't wrap my head around the proper synchronization and I got mediocre results :D I am willing to spend some effort on a web UI (seems more practical) for go-ios though. I don't see a use case for some of the available functionality but it can be useful for others. It will need a REST API as Daniel mentioned so it might be best to do it in parallel. @danielpaulus let me know if you are interested in scaling go-ios in such a way, I would love to contribute

**Comment by danielpaulus** [comment]
copy pasting what @LRF520 said:
I found an instance of GitHub, which I have successfully built locally and can run. Unfortunately, it only has Windows version, not MacOS version, so I think many people's projects are only half done, and do not really cover the user needs of a variety of systems. But I think go IOS has a greater influence, and many similar tools are actually based on go IOS, so I think if you build this general tool for windows and MacOS, it should help many people in need in the world, because no one in the whole world makes the interface of this tool.

I think it would be better if we could use go IOS to make a tool similar to scrcpy, make it suitable for windows and MacOS, and use USB cable to display the iPhone screen on the screen suitable for windows and MacOS!

I've tried your QuickTime_ video_ Hack, but I think QuickTime_ video_ Hack would be better if it could be packaged into an installation package. Unfortunately, it can't run on Windows system, so I always think that only half of the good tools are not perfect. Because I use windows and MacOS at the same time every day, I can't find a complete tool.

Instance:

(1)https://github.com/yhforthefreedom/iOSToolBox
(2)https://github.com/Genymobile/scrcpy

**Comment by shamanec** [comment]
I've already done simple remote control in my own project where I heavily utilize Appium and WebDriverAgent for that (which I can install and run in a Docker container thanks to `go-ios` <3). I can see benefit from a way to do stuff without using a terminal for some of the functionality, especially if managing multiple devices. Don't know if the general public of `go-ios` actually needs something like that though
